### ISSUE TITLE
scss-lint:disable is leaking into other files.

### ISSUE THREAD

**Issue by pselden** [brigade/scss-lint#443]
I started to integrate scss-lint into one of my pre-existing projects and marked several of my files with ```//scss-lint:disable all``` so I could tackle them later.

I run it with a rather large list of files, and I noticed that some files are not being flagged for lint errors even when they should be. I realized when I removed scss-lint:disable from a different file, it then flagged the errors in the other file correctly.

I have tried adding //scss-lint:enable all to that file but it still does nothing. The strange part is that when I remove all of the styles from the file that contains the disable comment (but leave the disable comment in), it then correctly flags the other file for errors.

I thought this might indicate a race condition (some shared flag is set but gets applied to a different file) but I don't see any multi-threading going on in the engine.

Any ideas or workarounds? If you have any suggestions about why this might be happening i can try to fix it.

**Comment by sds** [comment]
See #429.

**Comment by pselden** [comment]
Thank you! Missed that issue!
